src/fwriteR.c

```diff
@@ -5,18 +5,23 @@
 #define DATETIMEAS_EPOCH     2
 #define DATETIMEAS_WRITECSV  3
 
+#define NEED2NATIVE(s) !(IS_ASCII(s) || (s)==NA_STRING)
```

Comment:
minor stuff...
I think checking for NA is cheaper than for ASCII so changing order of that check could give some speed up
```
(s)!=NA_STRING && !IS_ASCII(s)
```